### ISSUE TITLE
[$250] QBD - Auto-sync toggle does not work correctly

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#52132]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.58-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5184320
**Email or phone of affected tester (no customers):** applausetester+dqbd7@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Precondition: QBD connection is established in the workspace.
1. Navigate to Accounting
2. Click on Advanced
3. disable Auto-sync toggle
4. Go back to Accounting and wait for the sync to finish return to Advanced

## Expected Result:
Auto-sync toggle is disabled

## Actual Result:
Auto-sync toggle returns to enabled after sync finishes

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/8a4db1da-38a1-4569-a495-e6ba63b26be2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021856102760605829790</li>
        <li>Upwork Job ID: 1856102760605829790</li>
        <li>Last Price Increase: 2024-11-11</li>
<li>Automatic offers: </li>
<ul>
<li>suneox | Reviewer | 104920810</li>
<li>ikevin127 | Contributor | 104999271</li>
</ul></li>
    </ul>
</details>

**Comment by johncschuster** [comment]
Triaged

**Comment by ikevin127** [comment]
I was able to reproduce / the issue happens on QBO as well. I will take over the issue from @suneox as C+ reviewer, coming from this Slack [thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1731680000279509).

##

Let's go with @narefyev91 from expert contributor group as they've shown interest in fixing the issue!

🎀👀🎀 C+ reviewed

**Comment by aldo-expensify** [comment]
Have we confirmed this a frontend issue?

**Comment by ikevin127** [comment]
> Have we confirmed this a frontend issue?

@aldo-expensify Looked into it and here's what happens:

When we turn **Auto-sync** off `UpdateQuickbooksDesktopAutoSync` is called -> returns response:

```json
{
    "jsonCode": 200,
    "requestID": "8e31e2695c68cf1b-SJC",
    "onyxData": [
        {
            "key": "policy_E12DF2A9245F1530",
            "onyxMethod": "merge",
            "value": {
                "connections": {
                    "quickbooksDesktop": {
                        "config": {
                            "autoSync": {
                                "enabled": false,
                                "jobID": null
                            }
                        }
                    }
                }
            }
        }
    ],
    "previousUpdateID": 2960739932,
    "lastUpdateID": 2960747654
}
```

After QBD / QBO syncs, FE calls `GetMissingOnyxMessages` automatically -> which returns the following response that re-enables **Auto-sync**:

```json
{
    "onyxData": [
        {
            "key": "policy_E12DF2A9245F1530",
            "onyxMethod": "merge",
            "value": {
                "connections": {
                    "quickbooksDesktop": {
                        "config": {
                            "autoSync": {
                                "enabled": true
                            }
                        }
                    }
                }
            }
        },
        {
            "key": "policy_E12DF2A9245F1530",
            "onyxMethod": "merge",
            "value": {
                "connections": {
                    "quickbooksDesktop": {
                        "config": {
                            "autoSync": {
                                "jobID": "9086846025541357"
                            }
                        }
                    }
                }
            }
        },
        {
            "key": "userMetadata",
            "onyxMethod": "set",
            "value": {
                "accountID": 15752938,
                "email": "kevin.bader96+2@gmail.com",
                "freeTrial": false,
                "planType": "corporate",
                "role": "admin",
                "tryNewDotDismissed": false
            }
        }
    ],
    "lastUpdateID": "2960783676",
    "previousUpdateID": "2960782355",
    "jsonCode": 200,
    "requestID": "8e31e7992b2f2536-SJC"
}
```

Looks like the first API call does not seem to persist the autoSync on the BE side which might have something to do with the fact that `jobID` is `null` for some reason which might mean that we don't pass it from FE when we make the call. What do you think ?

**Comment by aldo-expensify** [comment]
> Looks like the first API call does not seem to persist the autoSync on the BE side which might have something to do with the fact that jobID is null for some reason which might mean that we don't pass it from FE when we make the call. What do you think ?

I don't think the frontend needs to send `jobID: null`, that shouldn't matter. There is some special code in the backend handling the job deleting and making that key null.

The steps described in the issue look incomplete

> 1. Navigate to Accounting
> 2. Click on Advanced
> 3. disable Auto-sync toggle
> 4. Go back to Accounting and wait for the sync to finish return to Advanced

I think first we should clear that up. Which case is it?


### Case 1

1. Navigate to Accounting
2. Trigger the policy sync manually (NEW STEP!)
3. Click on Advanced
4. disable Auto-sync toggle
5. Go back to Accounting and wait for the sync to finish return to Advanced

### Case 2

1. Navigate to Accounting
3. Click on Advanced
4. disable Auto-sync toggle
5. Go back to Accounting
6. Trigger the policy sync manually (NEW STEP!)
7. Wait for the sync to finish return to Advanced


What I'm trying to see is what is trigger the sync

**Comment by aldo-expensify** [comment]
From looking at the video, I don't understand what triggered the sync in the first place. It seems to have happened after closing the Advanced settings modal... is that a bug? why did it sync?

**Comment by ikevin127** [comment]
> From looking at the video, I don't understand what triggered the sync in the first place. 

@aldo-expensify Looks like when `Auto-sync` is turned off, this starts a QBD sync process -> shortly after sync is completed -> FE calls `GetMissingOnyxMessages` which turns it back on.

Doesn't have anything to do with closing the Advanced settings modal, the `GetMissingOnyxMessages` will happen regardless:

https://github.com/user-attachments/assets/149cd8bd-ce6e-4014-9069-b3823008f09f

**Comment by aldo-expensify** [comment]
Thanks for checking that. For the moment, I have no idea where the problem is (BE/FE), I can try to reproduce / debug on Monday.

**Comment by suneox** [comment]
@ikevin127 will take over on [this issue](https://github.com/Expensify/App/issues/52132#issuecomment-2479684730) so i'd like unassign on this one

**Comment by narefyev91** [comment]
Yeah - after my checks in FE - i do not see anything strange in how we interact with that toggle. It's just pick value from Onyx data

**Comment by aldo-expensify** [comment]
> I was able to reproduce / the issue happens on QBO as well. I will take over the issue from @suneox as C+ reviewer, coming from this Slack [thread](https://expensify.slack.com/archives/C02NK2DQWUX/p1731680000279509).

@ikevin127 is there anything special to do to reproduce? I'm not being able to reproduce using QBO in dev. 🙇

**Comment by aldo-expensify** [comment]
Added @ikevin127 as C+

**Comment by ikevin127** [comment]
> [!tip]
> ### Steps on my side
> 1. Have QBD or QBO connected to a WS.
> 2. Go to Accounting > QBD / QBO > Advanced.
> 3. Auto-sync should be on by default -> turn it off.
> 4. Wait long enough with Advanced RHP panel opened and observe the following things happening:
> - Auto-sync toggle stays off (for now)
> - QBD / QBO connection will start to sync (see secondary text under connection name changing)
> - couple of seconds after sync completed (5-30 depends), the Auto-sync toggle will be turned back on automatically (if Networking console is opened you'll notice this happens after FE calls `GetMissingOnyxMessages`)

@aldo-expensify This is what's happening on my side (dev) for QBD:

<details>
<summary>Dev (QBD)</summary>

https://github.com/user-attachments/assets/e39df6d2-f582-402b-a1ac-9b24b57a4fdc

</details>

and this also on dev for QBO:

<details>
<summary>Dev (QBO)</summary>

https://github.com/user-attachments/assets/2c1edc3e-5617-4c1d-a7a9-dd95b8f3e063

</details>

##

Looks like `UpdateQuickbooksOnlineAutoSync` is taking unexpectedly long to complete on dev but anyway, after that completes it looks like the connection auto-sync doesn't automatically trigger anymore right away, but if I manually auto-sync, upon completion of the sync the `GetMissingOnyxMessages` is called which turns the Auto-sync toggle back to on.

Here's how it looks on staging (same thing):

<details>
<summary>Staging (QBO)</summary>

https://github.com/user-attachments/assets/2c1edc3e-5617-4c1d-a7a9-dd95b8f3e063

</details>

##

While all 3 videos above are pretty much the same, what I noticed is that even if we simply turn Auto-sync off and wait long enough, connection auto-sync will happen, then `GetMissingOnyxMessages` will get called which will turn Auto-sync toggle it back on:

https://github.com/user-attachments/assets/652c1a33-57ba-44d5-970b-8fe676c5ac7c

**Comment by ikevin127** [comment]
Not overdue, we're sorting out the issue.

cc @aldo-expensify for feedback on https://github.com/Expensify/App/issues/52132#issuecomment-2489972602.

**Comment by johncschuster** [comment]
Bumping for Melvin

**Comment by aldo-expensify** [comment]
Sorry, I haven't investigated again this. I'll give it some time tomorrow since I'm at the end of my day today.

**Comment by ikevin127** [comment]
Not overdue, we're sorting out the issue.

**Comment by aldo-expensify** [comment]
I spent some time trying to debug this, but I have never seen the `autoSync` going back to enabled unless I touch it while a sync triggered by another field is still ongoing.

I even saw a `GetMissingUpdates` request after disabling the `autoSync`, but still the updates there didn't switch back the setting:

<img width="1700" alt="image" src="https://github.com/user-attachments/assets/150401f0-b324-448b-8e2c-44b04c58dfda">

**Comment by aldo-expensify** [comment]
What I guess may be happening is something like this:

1. Sync gets triggered somehow
2. Integration server (IS) loads the policy data and starts communicating with QBO
3. User changes a setting in NewDot (i.e. `Auto sync`). This queues a new sync waiting for the current one to finish.
4. The sync that started in step 1 finishes and sends a request the PHP to save changes. The "changes" are outdated and don't include the latest setting value the user changed in step 3.
5. The server saves the changes to the database and pushes updates to the clients.
6. The user see the setting reverting

**Comment by ikevin127** [comment]
@aldo-expensify Just retested and the issue is still reproducible on both QBO and QBD, tested both on dev / staging API. 

Once the `Auto-sync` is turned off and the connection sync is completed -> if we wait long enough `GetMissingOnyxMessages` call is triggered (not sure by what) the `Auto-sync` will be turned back on.

https://github.com/Expensify/App/blob/5f3782b025b78e05516de86eec5a9cb2a0b3e01c/src/libs/actions/App.ts#L326-L330

`GetMissingOnyxMessages` is called on FE side in two places related to `OnyxUpdateManager`:
- [`handleOnyxUpdateGap`](https://github.com/Expensify/App/blob/5f3782b025b78e05516de86eec5a9cb2a0b3e01c/src/libs/actions/OnyxUpdateManager/index.ts#L72)
- [`validateAndApplyDeferredUpdates`](https://github.com/Expensify/App/blob/5f3782b025b78e05516de86eec5a9cb2a0b3e01c/src/libs/actions/OnyxUpdateManager/utils/index.ts#L116)

There were instances where `GetMissingOnyxMessages` was triggered shortly after connection sync is completed, and I also had instances where I had to wait 4-5 minutes for it to trigger, but when it does, it always turns `Auto-sync` back on.

Here are the details you asked for from my side - `UpdateQuickbooksOnlineAutoSync`:
```js
{
    policyID: 18920FC1A9030899,
    email: "kevin.bader96@gmail.com",

    requestID: "8edefd622fd89648-SJC",
    lastUpdateID: 3280895077,
    previousUpdateID: 3269474291,
}
```
then after the one above, `GetMissingOnyxMessages` was called with this data ~3 minutes later:
```js
{
    requestID: "8edf0bccaec123a9-SJC",
    lastUpdateID: "3280898069",
    previousUpdateID: "3280895077",
}
```

which turned `Auto-sync` back on as a result.

**Comment by aldo-expensify** [comment]
Thanks @ikevin127 , I'll investigate again later this week with the information you provided 🙇

**Comment by johncschuster** [comment]
How's this going, @aldo-expensify?

**Comment by aldo-expensify** [comment]
Thanks for the ping.. investigating now:

- Logs for when @ikevin127 disabled auto sync: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%228edefd622fd89648-SJC%22)+AND+timestamp:[2024-12-06T19:22:44.890Z+TO+2024-12-06T21:22:44.890Z]&index=logs_expensify-031551

I can confirm that the setting is being set to `false` in that request (disabling autoSync). I also see that since a setting was changed, we queue a job to run the policy integration sync.

- Logs for the integration policy sync after the auto sync changed: https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%22h92cWx%22)+AND+timestamp:[2024-12-06T19:22:45.629Z+TO+2024-12-06T21:22:45.629Z]&index=logs_expensify-031551

In those logs, I can see that the Integration Server is turning back on the autoSync setting. The only place that can happen for QBO I believe is here:

https://github.com/Expensify/Integration-Server/blob/c89f2fb0f140701543779a6069b309cf4f5af6b7/src/expensify/quickbooksonline/QuickbooksOnlineImporter.java#L217

So that narrows down the problem to figuring out why the Integration Server is considering that the policy has never been configured before.

**Comment by aldo-expensify** [comment]
I'm trying to debug this further, but trying to connect a new QBO accounts redirects me to an invalid URL in dev, I don't know if there is a new bug or something else:

```
https://integrations.expensify.com.dev/Integration-Server/null?client_id=Q00XcQZaPUmMXcob
```

<img width="860" alt="image" src="https://github.com/user-attachments/assets/6def18fb-9b2c-480c-8a81-156e4448cb51" />

**Comment by ikevin127** [comment]
@aldo-expensify This is a known issue, but you ca get around it by simply removing the `.dev` part from the link and proceed as usual.

**Comment by aldo-expensify** [comment]
I'm pretty sure the issue is not the `.dev`, since I'm on purpose trying to work against the integration server running in my dev environment. The issue seems to be that `/null` which I'm not sure where is it coming from (the code has changed a lot since the last time I worked on this 🥲 ).

**Comment by aldo-expensify** [comment]
There seems to be a problem with the Integration Server in my dev env, since the URL that ends up with `/null` is being constructed there

**Comment by aldo-expensify** [comment]
I'm still trying to fix my dev environment to work on this, I'm asking here: https://expensify.slack.com/archives/C3BGN04K1/p1734356547490549

**Comment by aldo-expensify** [comment]
I was just able to fix my dev environment, so I'll be able to resume my investigation / debugging today

**Comment by aldo-expensify** [comment]
@ikevin127 did you add the QBO connection from Old Dot? or New Dot? I didn't find in the logs a call from your account to `ConnectPolicyToQuickbooksOnline` so that makes me thing you connected in Old Dot and then switched to New Dot. Is that correct?

**Comment by ikevin127** [comment]
No idea whether the very first time was from OD, but after that I disconnected and re-connected multiple times all from ND.

**Comment by aldo-expensify** [comment]
> No idea whether the very first time was from OD, but after that I disconnected and re-connected multiple times all from ND.

I'm referring to this specific time you tested: https://github.com/Expensify/App/issues/52132#issuecomment-2524177414 . If you connected to QBO that same day for testing, from looking at logs, I think it wasn't through New Dot.

**Comment by aldo-expensify** [comment]
So far the only possible way I see for this to happen is:

1. User connects QBO in OldDOt
2. OldDot shows the configuration modal but the user doesn't save (just close the modal or the tab). When this happens, a configuration is not saved and `connections.quickbooksOnline.config.lastConfigurationTime` is missing.
3. User goes to New Dot and changes a QBO setting
4. The sync job runs because a setting was changed
5. The IS finds a missing `connections.quickbooksOnline.config.lastConfigurationTime`, so it assumes you are just connecting QBO for the first time, so it sets all the default values plus `autoSync: true`.

This sounds unlike to happen in a real case, so lowering the priority. I'm not sure about what we should do about it.

**Comment by ikevin127** [comment]
Makes sense, let me try this on a fresh account with connecting directly from ND to make sure I didn't go through OD flow initially, then and check whether the issue is still reproducible - will pass the details regarding policyID and all other from the fresh try.

**Comment by ikevin127** [comment]
Did as mentioned in the comment above and the issue is not reproducible if the connection flow is done strictly on ND, here are the details from `UpdateQuickbooksOnlineAutoSync`:

```js
{
    policyID: C6E8EC51B5025647,
    email: "kevin.bader96+397@gmail.com",
    requestID: "8f44e0e189a9cf1b-SJC",
    lastUpdateID: 3487737068,
    previousUpdateID: 3487736201,
}
```

Tried both with test emails (using +) and non-test and was not able to reproduce when connection was made from ND.

**Comment by aldo-expensify** [comment]
I think what @parasharrajat mentioned happens because we abort setting the default configuration if those accounts are missing:

https://github.com/Expensify/Integration-Server/blob/e3701f9f6b147902477e441d832225f1441c5bbf/src/expensify/quickbooksonline/QuickbooksUtils.java#L489-L492

I'm not sure what to do about it. Maybe we should still set the `lastConfigurationTime` to avoid the `autoSync` revert bug in this case. cc @francoisl since we worked on this early return here: https://github.com/Expensify/Integration-Server/pull/8126/files

**Comment by francoisl** [comment]
`lastConfigurationTime ` only applies to Quickbooks Online, right? As far as I can tell, it doesn't exist for Quickbooks Desktop  nor the other accounting integrations.

**Comment by francoisl** [comment]
TBH I don't know. Looking at the original Web-E code from 2016, looks like it was meant to signify that the policy had been "configured", i.e. the user had opened the config modal and then clicked Save. It also doesn't even seem like we use that value in NewDot.

**Comment by aldo-expensify** [comment]
> TBH I don't know. Looking at the original Web-E code from 2016, looks like it was meant to signify that the policy had been "configured", i.e. the user had opened the config modal and then clicked Save. It also doesn't even seem like we use that value in NewDot.

Considering that, I'm not sure what to do really, maybe the best would be to:

- Set as many default values as possible here https://github.com/Expensify/Integration-Server/blob/e3701f9f6b147902477e441d832225f1441c5bbf/src/expensify/quickbooksonline/QuickbooksUtils.java#L489-L493, and set the `lastConfigurationTime` to the current time, even if all those accounts are missing.

**Comment by aldo-expensify** [comment]
> Does that mean we'd never enable auto-sync for new connections though?

I don't think it means that. For auto-sync, It would just get rid of the effect of auto-sync returning to enabled all the time.

The other change this would bring is:

1. Connect QBO that doesn't have the minimum accounts: we will set the `lastConfigurationTime`
2. User adds the accounts on the QBO side of things
3. Some configuration change on NewDot causes a sync: we won't try to set the default settings again

The current behaviour is:

1. Connect QBO that doesn't have the minimum accounts: we don't set `lastConfigurationTime`
2. User adds the accounts on the QBO side of things
3. Some configuration change on NewDot causes a sync: since `lastConfigurationTime` is not set, we will now set the default values using the accounts

I think this change is fine

> Another consideration is the recent "avoid storing default values" general guideline. We could instead fall back to just reading the first value of the account lists in data during export, but I think it won't solve the issue here.

I think this could mean lots of changes that can risk a bug and is not worth the effort here. In `reportNameValuePairs` it was worth it because it saved us from storing millions of RNVPs, but I don't think we would have a similar return here.

**Comment by ikevin127** [comment]
@aldo-expensify 👋 What's the next step here: closing as 'not worth the effort' or we're looking at possibly updating the **Expected result** in order to open this up for proposals ?

**Comment by aldo-expensify** [comment]
> [@aldo-expensify](https://github.com/aldo-expensify) 👋 What's the next step here: closing as 'not worth the effort' or we're looking at possibly updating the **Expected result** in order to open this up for proposals ?

I'll put up a backend PR to implement this: https://github.com/Expensify/App/issues/52132#issuecomment-2592919363

**Comment by aldo-expensify** [comment]
@parasharrajat or @ikevin127, do you have some tips on how to setup a new QBO account for testing? I want to set up an account that is very incomplete to reproduce what @parasharrajat had here: https://github.com/Expensify/App/issues/52132#issuecomment-2565378939

**Comment by johncschuster** [comment]
@aldo-expensify were you able to set up that QBO account? Can I help with that?

**Comment by francoisl** [comment]
What's the doc that explains how to connect? Does [this guide](https://stackoverflowteams.com/c/expensify/questions/5030/16010#16010) work instead?

**Comment by aldo-expensify** [comment]
> What's the doc that explains how to connect? Does [this guide](https://stackoverflowteams.com/c/expensify/questions/5030/16010#16010) work instead?

This [doc ](https://docs.google.com/document/d/1MgDyAUG-WaW-fxic_Uhh-HkbMWkKzCXUaCZdwQkGLnQ/edit?tab=t.0) (I got access now). I had not seen that SO either, so when I have some time I'll try both (whatever works :))

**Comment by ikevin127** [comment]
Not overdue, we're still looking into this from the BE side.

**Comment by johncschuster** [comment]
@aldo-expensify can we get an update on this? How's this going?

**Comment by aldo-expensify** [comment]
I'll test now.. I have been focusing on things that are higher priority, this looks like a low priority edge case to me

**Comment by aldo-expensify** [comment]
Hmm, I'm stuck getting this error when I try to stablish a new connection:

https://github.com/Expensify/Web-Expensify/pull/43700/files#diff-5d6a98f86e92ee84dc7ee054463a257819e80d8673978c73a9ae80eebdeda859R47-R50

@lakchote do you know what I may be missing in my environment that is making the `message = ''` in IS's response? 🙇

**Comment by lakchote** [comment]
> Hmm, I'm stuck getting this error when I try to establish a new connection:
> 
> do you know what I may be missing in my environment that is making the `message = ''` in IS's response? 🙇
>

Is the error related to the `setupURL` or is `message` empty and throwing `Missing message from integration server`?

If `message` is empty that's definitely weird. I have looked into production and it didn't happen.

Could you please check your logs and post what you have, looking for `Creating a new Codat Company` (https://github.com/Expensify/Integration-Server/blob/main/src/expensify/codat/quickbooksdesktop/QuickbooksDesktopConnect.java#L54) up until `Generated setup URL` please? Thank you!

**Comment by aldo-expensify** [comment]
> Is the error related to the setupURL or is message empty and throwing Missing message from integration server?

The message is an empty string, but it does exists, so I'm not getting the `Missing message from integration server` error... I'm getting the `Missing setupUrl from integration server ` error

> Could you please check your logs and post what you have?

I'll check again.. I'm pretty sure the logs from the integration server were almost empty

**Comment by aldo-expensify** [comment]
> > Could you please check your logs and post what you have?
> 
> I'll check again.. I'm pretty sure the logs from the integration server were almost empty

I can see the log line: `Running QuickbooksDesktopConnect job for policy`, but I don't see `Creating a new Codat Company`... so I think it is crashing before. I'll debug it more...

**Comment by aldo-expensify** [comment]
Ahh, seems like I'm missing my java keystore... maybe that will fix it...

**Comment by lakchote** [comment]
> I can see the log line: Running QuickbooksDesktopConnect job for policy, but I don't see Creating a new Codat Company...

That's definitely unusual.

I've checked in the Codat logs for failed connections and there is nothing:

![Image](https://github.com/user-attachments/assets/2523bf03-2e4e-4dc8-ac57-9e899157b503)

Could you please try something?
Before launching the QBD connection process in NewDot, log in first on OldDot with the same account.
Then try to connect to QBD.

Also cc @francoisl for external eyes on this too

**Comment by francoisl** [comment]
I think I figured out the issue, sent your a PR - https://github.com/Expensify/Integration-Server/pull/8344

**Comment by lakchote** [comment]
> I think I figured out the issue, sent your a PR - [Expensify/Integration-Server#8344](https://github.com/Expensify/Integration-Server/pull/8344)

Did this fix it for you @aldo-expensify?

**Comment by aldo-expensify** [comment]
> > I think I figured out the issue, sent your a PR - [Expensify/Integration-Server#8344](https://github.com/Expensify/Integration-Server/pull/8344)
> 
> Did this fix it for you [@aldo-expensify](https://github.com/aldo-expensify)?

I haven't been able to reproduce it, but the change makes sense to me.

We should ask someone that can reproduce this to retest after https://github.com/Expensify/Integration-Server/pull/8344 has been deployed.

**Comment by francoisl** [comment]
Deploying it to staging now, it should be available for testing in a few minutes.

**Comment by aldo-expensify** [comment]
@lanitochka17 can you retest this in staging? 🙏

**Comment by aldo-expensify** [comment]
Friendly bump @lanitochka17 , can we get some help with retesting this please 🙏

**Comment by izarutskaya** [comment]
Issue is not reproducible

https://github.com/user-attachments/assets/d8578725-f170-4e1f-acd1-b2a65e13e0cc

**Comment by aldo-expensify** [comment]
Thanks!, I'll close then
